documentation/modules/deploying/proc-deploy-migrate-kraft.adoc

```diff
@@ -0,0 +1,257 @@
+The name of the controller node pool is `controller`.
+
+.Procedure
+
+. For the Kafka cluster, create a node pool with a `controller` role.
++
+Add a quorum of controller nodes to the node pool.
++
+.Example configuration for a controller node pool
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaNodePoolApiVersion}
+kind: KafkaNodePool
+metadata:
+  name: controller
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 20Gi
+        deleteClaim: false
+    resources:
+      requests:
+        memory: 64Gi
+        cpu: "8"
+      limits:
+        memory: 64Gi
+        cpu: "12"    
+----
++
+NOTE: For the migration, you cannot use a node pool of nodes that share the broker and controller roles.
+
+. Apply the new `KafkaNodePool` resource to create the controllers.
++
+Errors related to using controllers in a ZooKeeper-based environment are expected.
+
+. Enable KRaft migration in the `Kafka` resource by setting the `strimzi.io/kraft` annotation to `migration`:
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: migration
+----
++
+.Enabling KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: migration 
+# ...
+----
++
+Applying the annotation to the `Kafka` resource configuration starts the migration.
+
+. Check the controllers have started and the brokers have rolled:
++
+[source,shell]
+----
+kubectl get pods -n my-project
+----
++
+.Output shows nodes in broker and controller node pools
+[source,shell]
+----
+NAME                     READY  STATUS   RESTARTS
+my-cluster-broker-0      1/1    Running  0
+my-cluster-broker-1      1/1    Running  0
+my-cluster-broker-2      1/1    Running  0
+my-cluster-controller-3  1/1    Running  0
+my-cluster-controller-4  1/1    Running  0
+my-cluster-controller-5  1/1    Running  0
+# ...
+----
+
+. Check the status of the migration:
++
+[source,shell]
+----
+kubectl get kafka my-cluster -n my-project -w
+----
++
+.Updates to the metadata state
+[source,shell]
+----
+NAME        ...  METADATA STATE   WARNINGS
+my-cluster  ...  Zookeeper
+my-cluster  ...  KRaftMigration
+my-cluster  ...  KRaftDualWriting
+----
++
+`METADATA STATE` shows the mechanism used to manage Kafka metadata and coordinate operations.
+At the start of the migration this is `ZooKeeper`.
++
+--
+* `ZooKeeper` is the initial state when metadata is only stored in ZooKeeper.
+* `KRaftMigration` is the state when the migration is in progress.
+* `KRaftDualWriting` is the state when the migration has ended and the cluster is working as a KRaft cluster. 
+Metadata is managed in Kafka but also replicated in ZooKeeper, so you can xref:proc-deploy-migrate-kraft-rollback-{context}[roll back from this point].
```

Comment:
Actually, along the way, we decided to not stop the migration at `KRaftDualWriting` as stable state but the next one which is the `KRaftPostMigration`.
It means that you have to expect the brokers rolling twice after the controllers are deployed.
The first time because the are configured to be connected to KRaft as well and the ZooKeeper migration flag is enabled. This way they can register to controllers and the migration starts.
Said that I would highlight that the migration can take long depending on how much topics and corresponding metadata need to be migrated. It means that watching the state could end being in the `KRaftMigration` state for long time (this is the actual state when the migration is happening). Checking that the migration is done or not is executed by the operator on reconciliation bases (so it means by default every 2 mins).
When the migration ends, the state machine moves to `KRaftDualWriting` and it triggers another roll of the brokers to remove the ZooKeeper connection and migration flag. So after rolling, the state moves to `KRaftPostMigration` waiting for the user. It's important to say that even in this state, the dual-write is happening.

Reply:
Thanks. I've updated as directed.